File: backend/src/main/java/com/pwr/it/app/data/domain/Organization.java
package com.pwr.it.app.data.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.UUID;

@Getter
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Organization {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private String fullName;
    private String phoneNumber;
    private String email;
    private String address;
    @JsonIgnore
    private String uuid = UUID.randomUUID().toString();

    public Organization(String fullName, String phoneNumber, String email, String address) {
        this.fullName = fullName;
        this.phoneNumber = phoneNumber;
        this.email = email;
        this.address = address;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        return uuid.equals(((Organization) o).uuid);
    }

    @Override
    public int hashCode() {
        return uuid.hashCode();
    }

}

File: frontend/src/models/Organization.ts
export class Organization {
    id: number = 0;
    phoneNumber: string = '';
    fullName: string = '';
    email: string = '';
    address: string = '';


	constructor(id: number, phoneNumber: string, fullName: string, email: string, address: string) {
		this.id = id;
		this.phoneNumber = phoneNumber;
		this.fullName = fullName;
		this.email = email;
		this.address = address;
	}

}
